### ISSUE TITLE
Avoid iOS manual configuration of AppDelegate required methods

### ISSUE THREAD

**Issue by Henry-Keys** [nextbss/flutter-push-notification#19]
Instead of forcing users to override the methods required to make the plugin work, use **FlutterPlugin** protocol and implement those methods in **SwiftFlutterPushNotificationPlugin** class.

Unfortunately, *application:didFailToRegisterForRemoteNotificationsWithError* is not in the **FlutterPlugin** protocol, so it must be overridden in the AppDelegate class.

**Comment by Henry-Keys** [comment]
Also, for some reason, *application:didReceiveRemoteNotification:fetchCompletionHandler* is not being called when a remote notification arrives. The app logs the message below:
```
Can't end BackgroundTask: no background task exists with identifier 5 (0x5), or it may have already been ended. Break in UIApplicationEndBackgroundTaskError() to debug.
```

*application:didReceiveRemoteNotification* is called instead and gives the crash related to issue https://github.com/nextbss/flutter-push-notification/issues/18.

I'll solve both issues with just one pull request

**Comment by Henry-Keys** [comment]
Fixed by #20
